### PR TITLE
fix(ingest)- airflow: update subdag check for compatibility with older Airflow versions

### PR DESCRIPTION
**Pull request by vcs9**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub_provider/client/airflow_generator.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def _get_dependencies(
             upstream_task = dag.task_dict[upstream_task_id]
 
             # if upstream task is not a subdag, then skip it
-            if upstream_task.subdag is None:
+            if getattr(upstream_task, 'subdag', None) is None:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub_provider/client/airflow_generator.py:35]:
```suggestion
            if getattr(upstream_task, "subdag", None) is None:
```

File: metadata-ingestion/src/datahub_provider/client/airflow_generator.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def _get_dependencies(
             [
                 DataJobUrn.create_from_ids(job_id=task_id, data_flow_urn=str(flow_urn))
                 for task_id in task.upstream_task_ids
-                if dag.task_dict[task_id].subdag is None
+                if getattr(dag.task_dict[task_id], 'subdag', None) is None
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub_provider/client/airflow_generator.py:116]:
```suggestion
                if getattr(dag.task_dict[task_id], "subdag", None) is None
```

**Review by hsheth2 — COMMENTED** [review]
Looks like the formatter wants double quotes :)

What version of airflow does this ensure compatibility with? Do the previous versions of airflow keep subdag information under a different attribute?

**Comment by vcs9** [comment]
> Looks like the formatter wants double quotes :)
> 
> What version of airflow does this ensure compatibility with? Do the previous versions of airflow keep subdag information under a different attribute?
This is for `1.10.3`. I'm not sure actually.. From the Airflow documentation, it looks like subdags were introduced before this version. The attribute `is_subdag` didn't through any errors.

**Review by hsheth2 — APPROVED** [review]
Looked into it a bit more - in newer versions of airflow, the BaseOperator sets `self.subdag = None` so that the attribute always exists, but this code was not present in previous versions like 1.10.3.

Ref: https://airflow.apache.org/docs/apache-airflow/1.10.15/_modules/airflow/models/baseoperator.html#BaseOperator
```
        # subdag parameter is only set for SubDagOperator.
        # Setting it to None by default as other Operators do not have that field
        self.subdag = None   # type: Optional[DAG]
```

vs https://airflow.apache.org/docs/apache-airflow/1.10.3/_modules/airflow/models.html#BaseOperator does not have that code.

**Pull request merged**